test/com/serotonin/mango/rt/dataSource/meta/MetaPointLocatorRTInitializeTest.java

```diff
@@ -0,0 +1,101 @@
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.modules.junit4.PowerMockRunnerDelegate;
+import utils.mock.MockUtils;
+
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+
+
+@RunWith(PowerMockRunner.class)
+@PowerMockRunnerDelegate(Parameterized.class)
+@PrepareForTest({MetaPointLocatorRT.ScheduledUpdateTimeout.class, MetaPointLocatorRT.class})
+// resources/org/powermock/extensions/configuration.properties is not working
+@PowerMockIgnore({"com.sun.org.apache.xerces.*", "javax.xml.*", "org.xml.*", "org.w3c.*", "com.sun.org.apache.xalan.*",
+        "javax.activation.*", "javax.management.*"})
+public class MetaPointLocatorRTInitializeTest {
+
+    @Parameterized.Parameters(name= "{index}: Update event type: {0}, Expected times of invocation: {1}")
+    public static Object[] data() {
+        return new Object[] {
+                new Object[] {MetaPointLocatorVO.UPDATE_EVENT_CONTEXT_CHANGE, 1},
+                new Object[] {MetaPointLocatorVO.UPDATE_EVENT_CONTEXT_UPDATE, 0},
+                new Object[] {MetaPointLocatorVO.UPDATE_EVENT_CRON, 0},
+                new Object[] {Common.TimePeriods.MINUTES, 0},
+                new Object[] {Common.TimePeriods.HOURS, 0},
+                new Object[] {Common.TimePeriods.DAYS, 0},
+                new Object[] {Common.TimePeriods.WEEKS, 0},
+                new Object[] {Common.TimePeriods.MONTHS, 0},
+                new Object[] {Common.TimePeriods.YEARS, 0},
+        };
+    }
+
+    private final DataPointRT dataPoint = mock(DataPointRT.class);
+    private final MetaDataSourceRT dataSource = mock(MetaDataSourceRT.class);
+
+    private final int expectedInvocationTimes;
+    MetaPointLocatorRT metaPointLocatorRT;
+    RealTimeTimer timer = mock(RealTimeTimer.class);
+
+    public MetaPointLocatorRTInitializeTest (int updateEvent, int expectedInvocationTimes){
+        this.expectedInvocationTimes = expectedInvocationTimes;
+
+        MetaPointLocatorVO metaPointLocatorVO = new MetaPointLocatorVO();
+        metaPointLocatorVO.setUpdateEvent(updateEvent);
+        metaPointLocatorVO.setDataTypeId(DataTypes.NUMERIC);
+        metaPointLocatorVO.setScript("return 1;");
+
+        if(metaPointLocatorVO.getUpdateEvent() == MetaPointLocatorVO.UPDATE_EVENT_CRON) {
+            metaPointLocatorVO.setUpdateCronPattern("30 30 12 ? * * *");
+        }
+
+        metaPointLocatorRT = new MetaPointLocatorRT(metaPointLocatorVO);
+
+        if(metaPointLocatorVO.getUpdateEvent() != MetaPointLocatorVO.UPDATE_EVENT_CONTEXT_CHANGE
+                && metaPointLocatorVO.getUpdateEvent() != MetaPointLocatorVO.UPDATE_EVENT_CONTEXT_UPDATE) {
+            timer.init();
```

Comment:
As I explained, this piece of code should never be executed, it causes the test to behave differently each time.